x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx

```diff
@@ -104,8 +105,10 @@ export function LensEditConfigurationFlyout({
             return customIsEqual(
               previousAttrs.state.visualization,
               previousAttrs.references,
+              previousAttrs.state.datasourceStates.formBased,
```

Comment:
```suggestion
              previousAttrs.state.datasourceStates,
```

Reply:
Fixed in 878a48556a848b8d34945cb7d90713e1a254928b